test/unit/server/simulate_transaction_test.js

```diff
@@ -235,3 +237,43 @@ function invokeSimulationResponseWithRestoration(address) {
     }
   };
 }
+
+describe('works with real responses', function() {
```

Comment:
not suggesting a change, just wondering on hardcoding xdr as precedence in tests or build the js xdr object graph and then serialize via `.toXdr()` rather than hardcoding?

Reply:
honestly I mostly just added this as a sanity check that the specific error we were seeing got fixed. another nice part about this is that if the XDR changes, this test will fail, prompting investigation and tests with a new RPC

I agree in general that it's better to build XDR then parse it rather than hardcoding, but that also doesn't reflect real RPC behavior sometimes.. we really need to build integration tests for RPC :arrow_right: soroban-client, but now isn't the time for that heh

Reply:
agreed, this isn't the place for that type of integration test, this should just work against the rpc interface, that's what system-test/e2e is for.